src/ajax/xhr.js

```diff
@@ -19,7 +19,19 @@ jQuery.ajaxSettings.xhr = window.ActiveXObject !== undefined ?
 			// and http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9
 			// Although this check for six methods instead of eight
 			// since IE also does not support "trace" and "connect"
-			/^(get|post|head|put|delete|options)$/i.test( this.type ) &&
+			//
+			// Support: IE 10-11
+			// IE seems to error on cross-domain PATCH requests when ActiveX XHR
+			// is used. Use the following strategy:
+			// 1. If request is local, use the ActiveX XHR.
+			// 2. Otherwise, in IE9+ always use the native XHR.
+			// 3. Otherwise, in IE8 if request type is *not* included in the following
+			//    list, use the ActiveX XHR.
+			// 4. Otherwise, use the native XHR.
+			// jshint -W018
+			( !( document.documentMode < 9 ) ||
+			// jshint +W018
+				/^(get|post|head|put|delete|options)$/i.test( this.type ) ) &&
```

Comment:
This all got quite convoluted. How about reversing the polarity?
```js
// Native XHR cannot access local files, always use ActiveX for that case
return this.isLocal ||

	// Also force ActiveX for non-RFC2616 methods with IE<9 (trac-13240)
	// http://msdn.microsoft.com/en-us/library/ie/ms536648(v=vs.85).aspx
	// ...but *ONLY* IE<9, since IE 10-11 cross-domain PATCH requires native XHR (gh-1684)
	document.documentMode < 9 && !/^(?:get|post|head|put|delete|options)$/i.test( this.type ) ?

	createActiveXHR() :
	createStandardXHR() || createActiveXHR();
```